out_textureblock.py

```diff
@@ -19,89 +27,227 @@ def shifter(a1):
+    t['format_a'] = get_value(Section5.getElementsByTagName("unk12")[0].attributes['u8'].value)
+    t['format_b'] = get_value(Section5.getElementsByTagName("unk13")[0].attributes['u8'].value)
+    t['flags'] = get_value(Section5b.getElementsByTagName("unk3")[0].attributes['u8'].value)
+
+    if texindex in texinfo:
+      if (texinfo[texindex] != t):
+        if history != '':
+          print('')
+        print("  Skipping conflicting texture information for %d" % texindex)
+        print("    Old: %s" % (str(texinfo[texindex])))
+        print("    New: %s" % (str(t)))
+        history = ''
+      else:
+        if history != 'skipping':
+          if history != '':
+            print('')
+          print("  Skipping known texture information for", end='')
+          history = 'skipping'
+        print(" %d" % texindex, end='', flush=True)
+    else:
+      if history != 'adding':
+        if history != '':
+          print('')
+        print("  Adding texture information for", end='')
+        history = 'adding'
+      print(" %d" % texindex, end='', flush=True)
+      texinfo[texindex] = t
+      texinfo_source[texindex] = path
+
+# Open the textureblock
 with open(sys.argv[1], 'rb') as f:
+
+  # Start to output tags for wxHexEditor
+  tags = open("/tmp/swep1r/wxhexeditor.tags", "w")
+  tags.write('<?xml version="1.0" encoding="UTF-8"?>\n' + '<wxHexEditor_XML_TAG>\n' + '  <filename>\n')
+  tag_id = 0
+  def add_tag(start, size, text, colour = "#888888", dump=False):
+    global tag_id
+    tag_id += 1
+    s = ''
+    s += '    <TAG id="%s">\n' % tag_id
+    s += '      <start_offset>%d</start_offset>\n' % start
+    s += '      <end_offset>%d</end_offset>\n' % (start + size - 1)
+    s += '      <tag_text>%s (%d bytes)</tag_text>\n' % (text, size)
+    s += '      <font_colour>#000000</font_colour>\n'
+    s += '      <note_colour>%s</note_colour>\n' % colour
+    s += '    </TAG>\n'
+    tags.write(s)
+    if dump:
+      off = f.tell()
+      f.seek(start)
+      buf = f.read(size)
+      with open('/tmp/swep1r/%s.bin' % text, 'wb') as t:
+        t.write(buf)
+      f.seek(off)
+
+  # Start dumping all textures
   count = read32(f)
-  for i in range(0, count - 1):
+  add_tag(0, 4, 'count', '#FF88FF')
+  add_tag(4 + 8 * count, 4, 'end-offset', '#8888FF')
+
+  for i in range(0, count):
+
     f.seek(4 + 8 * i)
-    a = read32(f)
-    b = read32(f)
-    c = read32(f)
-    length = c - a
+    off_a = read32(f)
+    off_b = read32(f)
+    off_c = read32(f)
+    length = off_c - off_a
+
+    add_tag(4 + 8 * i + 0, 4, 'pixel-offset-%d' % i)
+    add_tag(4 + 8 * i + 4, 4, 'palette-offset-%d' % i)
+    if off_b == 0:
+      add_tag(off_a, off_c - off_a, 'pixel-data-%d' % i, "#FFFF88", dump=False)
+    else:
+      add_tag(off_a, off_b - off_a, 'pixel-data-%d' % i, "#FF8888", dump=False)
+      add_tag(off_b, off_c - off_b, 'palette-data-%d' % i, "#88FF88", dump=False)
```

Comment:
nit: There are no tags for the actual used amount of data. I did not add that, as it would always require knowledge about the modelblock or the game binary (which know how to interpret these raw blocks of data).